src/lib/fetcher/fetcher.ts

```diff
@@ -0,0 +1,139 @@
+import { IRequestWrapper } from '../../dex-helper';
+import { Logger } from 'log4js';
+import { RequestConfig, Response } from '../../dex-helper/irequest-wrapper';
+
+import Timeout = NodeJS.Timeout;
+
+const FETCH_TIMEOUT_MS = 10 * 1000;
+const FETCH_FAIL_MAX_ATTEMPT = 5;
+const FETCH_FAIL_RETRY_TIMEOUT_MS = 60 * 1000;
+
+export type RequestInfo<T> = {
+  requestOptions: RequestConfig;
+  caster: (data: unknown) => T | null;
+  authenticate?: (options: RequestConfig) => void;
+  excludedFieldsCaching?: string[];
+};
+
+export type RequestInfoWithHandler<T> = {
+  info: RequestInfo<T>;
+  handler: (data: T) => void;
+};
+
+export default class Fetcher<T> {
+  private intervalId?: Timeout;
```

Comment:
Name is misleading. It should be changed to `timer` / `timeoutTImer` or whatever